### ISSUE TITLE
Errors don't clear up after being fixed

### ISSUE THREAD

**Issue by lanzz** [microsoft/vscode-mypy#139]
```python
from typing import Any

def foo(a: Any, b: Any) -> bool:
    return (a == b)
```
The above code fails with a "Returning Any from function declared to return "bool" (Mypy no-any-return)", which is understandable. The problem manifest once I fix it:
```python
from typing import Any

def foo(a: Any, b: Any) -> bool:
    return bool(a == b)
```
The return statement remains annotated with the same error, even though it is fixed now. The error only goes away if I restart VScode. I can also delete the entire line, save, then re-add it, which will fix the return type error, but will leave me with a different stale error: "expected an indented block after function definition on line 4 (Mypy syntax)", which was raised while the row was deleted.

I'm running pre-release version v2023.3.12421009 because of the `--show-error-code` bug still not fixed on the stable channel. There are no obvious problems logged in the Mypy Type Checker output stream.
